packages/manager/apps/pci-kubernetes/src/hooks/useFlavors.ts

```diff
@@ -175,7 +175,7 @@ export const useMergedKubeFlavors = (projectId: string, region: string) => {
           )?.category,
           isFlex: /flex$/.test(flavor.name),
           isLegacy: /eg|sp|hg|vps-ssd/.test(flavor.name),
-          hasEnoughQuota: hasEnoughQuota(flavor, quota[0]),
+          hasEnoughQuota: hasEnoughQuota(flavor, quota as TQuota),
```

Comment:
AFAIS, the `useProjectQuotaByRegion` hook returns `TQuota[]` if no select is provided. That is the cas in line 109 when you invokes it with no select parameter.

Here you type cast an array of quotas into a single object, that is absolutely not what you want here, because it means that `hasEnoughQuota` function manipulates an object (`TQuota`) and you give it an array (TQuota[]).

So it cannot work like this

Reply:
BTW prefer using type inference to be safer, here the type cast misled you.

Instead, you can say the `data` returned by the hook is either `TQuota[]`  or the select function's <ReturnType>